src/slotshandler.cpp

```diff
@@ -1347,20 +1352,10 @@ int tape_insert_voc (FILE *pfile)
 void cartridge_load (void)
 {
   if (CPC.model >= 3) {
-    if (!CPC.cart_file.empty()) { // load a cartridge ?
-      if (CPC.cart_zip) { // compressed image?
-        FILE *file = extractFile(CPC.cart_path, CPC.cart_file, ".cpr");
-        if (file) {
-          cpr_load(file);
-          fclose(file);
-        }
-      } else {
-        if(cpr_load(CPC.cart_path + CPC.cart_file) != 0) {
-          fprintf(stderr, "Load of cartridge failed. Aborting.\n");
-          exit(-1);
-        }
-      }
-    }
+     if (file_load(CPC.cart_path + CPC.cart_file, OTHER)) {
```

Comment:
why not just always set CPC.cart_file to the full path ? And use path only for Load/Save menu as for others.